metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -1283,6 +1222,64 @@ def get_tables_for_dataset(
 
         return bigquery_tables
 
+    def get_core_table_details(
+        self, conn: bigquery.Client, dataset_name: str, project_id: str
+    ) -> Dict[str, TableListItem]:
+        table_items: Dict[str, TableListItem] = {}
+        # Dict to store sharded table and the last seen max shard id
+        sharded_tables: Dict[str, TableListItem] = defaultdict()
```

Comment:
What does this being a `defaultdict` give us? The only default usage I see is `sharded_tables[table_name].table_id` but idk how this would work because the `default_factory` is `None` right? Just seems like it complicates the logic because we have to check:
`if not sharded_tables.get(table_name):` rather than `if table_name not in sharded_tables`

Reply:
makes sense, I can change it